src/api.ts

```diff
@@ -105,6 +166,29 @@ export class DXApiClient implements DXApi {
     return await resp.json();
   }
 
+  private async getFromApp<T = any>(
+    path: string,
+    params: Record<string, string | null | undefined>,
+  ): Promise<T> {
+    const proxyHost = `${await this.discoveryApi.getBaseUrl("proxy")}/dx-app`;
+
+    const url = new URL(`${proxyHost}${path}`);
+
+    for (const [key, value] of Object.entries(params)) {
+      if (value !== undefined && value !== null) {
+        url.searchParams.append(key, value);
+      }
+    }
+
+    const { fetch } = this.fetchApi;
+
+    const resp = await fetch(url, { method: "GET" });
```

Comment:
It would be nice to add some headers here to identify the client in our telemetry. We might do:

- `x-client-type: backstage-plugin`
- `x-client-version: <plugin-version>`
- Maybe something with `dx.appId`?

Reply:
Yes. I vote yes.